.github/workflows/build.yml

```diff
@@ -151,20 +151,20 @@ jobs:
           sparse-checkout: |
```

Comment:
I see... is it maybe dangerous for us to use sparse-checkout in CI at all? It seems like this introduces the possibility of differences between running jobs on CI and running things locally (differences that are really hard to spot/debug)? 

If we're going to optimize our CI for something, I'd rather go with reliability over speed (since fast, unreleable CI costs us lots of human time... which is way more expensive and constrained than github actions compute time for us).

Reply:
> It seems like this introduces the possibility of differences between running jobs on CI and running things locally (differences that are really hard to spot/debug)?

Yes, it does. The only alternative I can think of (that also makes sure all these .props and .targets and other files I can't even think of are not picked up from parent directories by dotnet tooling) is moving all the sources of the current repo to another directory and having the integration test as a sibling, not child of that root directory. 

Well no, not the only one, the other alternative is building in a directory outside the project root (e.g. in %TEMP%) which is not great for local development either because then you have a hard time manually checking the generated project files if something doesn't work.

They all have their costs ¯\_(ツ)_/¯

Reply:
I'm open to suggestions how to achieve the isolation during tests (at least in CI, as it is now) that would improve local development experience

Reply:
Should I update the comment in any way? That's what it was trying to get across so not sure if you've missed it (I usually ignore skip comments...) or it's unclear.

Reply:
I hadn't seen the comment actually (I was looking at the commit diff in github - not sure if that even showed). Maybe worth adding `(e.g. Directory.Build.props etc.)` just to be clear about what is meant by local files impacting integration tests.

Reply:
done, thx for the feedback.